decision-records/2024-06-04--service-to-stop-while-waiting-for-cert.md

```diff
@@ -0,0 +1,106 @@
+**Date:** 2024-06-04<br/>
+**Author:** @sed-i
+
+## Context and Problem Statement
+When an admin forms a tls relation, there is a time window in which:
+
+1. the juju model tells us we must operate with TLS enabled; however
+2. we haven't received the server cert yet.
```

Comment:
```suggestion
2. we haven't received the server cert yet (actual, transient state).
```

as I mention below, this isn't probably _just about_ the cert missing, but about whether all preconditions are met for enabling tls, including:
- the cert is on disk (charm and workload(s))
- update-ca-certificates --fresh has run and exited 0
- charm has privkey